### PR TITLE
Projects/better typer

### PR DESCRIPTION
**Pull request by lashlockbhs**

### CODE REVIEW

File: projects/BetterTyper/public/script.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+const arToStr = (ar, a) => {
+    let str = ""
+    for (let i = 0; i < ar.length; i++) {
+        str += ar[i].toStr() + a
+    }
+    return str;
+}
+
+const randVarDec = (names, calls, depth, tabs) => {
```
Comment (gigamonkey) [inline projects/BetterTyper/public/script.js:93]: One pattern you might want to consider is using `static` methods on the classes themselves to generate random instances of the class. Here's a trivial example:

```javascript
class Foo {
  static random() {
    return new Foo();
  }
  toString() {
    return "I'm a Foo!"
  }
}

class Bar {
  static random() {
    return new Bar();
  }
  toString() {
    return "I'm a Bar!"
  }
}

console.log(Foo.random());
console.log(Bar.random());
console.log((Math.random() < 0.5 ? Foo : Bar).random());
```

**Review by gigamonkey — APPROVED** [review]
Good start. Left some comments about various things.

File: projects/BetterTyper/public/script.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        this.ret = ret;
+        this.tabs = tabs;
+    }
+
+    toStr() {
+        return this.tabs + "const " + this.name + " = (" + arToStr(this.args, ", ") + ") => {" + "\n" + arToStr(this.values, "\n") + this.tabs + "   " + "return " + this.ret.toStr() + "\n" + this.tabs + "}"
+    }
+}
+
+class varDec {
```
[the remark this answers is not in the document]
Comment (lashlockbhs) [reply projects/BetterTyper/public/script.js:40]: Ill change it, Ik about the uppercase I just never got around to it

File: projects/BetterTyper/public/script.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+class varCall {
+    constructor(name) {
+        this.name = name;
+    }
+    toStr() {
+        return this.name;
+    }
+}
+
+class equation {
```
[the remark this answers is not in the document]
Comment (lashlockbhs) [reply projects/BetterTyper/public/script.js:62]: in this case it made sense cause you cant just throw a 2 + 2 = 4 in js, I feel like I would get confused if I named it differently. Also equation in this case is a value not a statment, which differs from what you would expect. But this just made sence to me.

File: projects/BetterTyper/public/script.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+const randEl = (ar) => ar[Math.floor(Math.random() * ar.length)];
+
+const arToStr = (ar, a) => {
+    let str = ""
+    for (let i = 0; i < ar.length; i++) {
+        str += ar[i].toStr() + a
+    }
+    return str;
+}
```
[the remark this answers is not in the document]
Comment (lashlockbhs) [reply projects/BetterTyper/public/script.js:91]: I was writing this in a rush just to stringify an array with comas, but join will work better thanks/

File: projects/BetterTyper/public/script.js (hunk 1 above)
Comment (lashlockbhs) [reply projects/BetterTyper/public/script.js:93]: this acually looks hella nice, more orgnzation, ill defiently impliment this

**Pull request merged**